src/nrnpython/nrnpython.cpp

```diff
@@ -45,16 +50,21 @@ int nrnpy_site_problem;
 
 void nrnpy_augment_path() {
   static int augmented = 0;
-  char buf[1024];
   if (!augmented && strlen(neuronhome_forward()) > 0) {
-    // printf("augment_path\n");
     augmented = 1;
-    sprintf(buf, "sys.path.append('%s/lib/python')", neuronhome_forward());
     int err = PyRun_SimpleString("import sys");
     assert(err == 0);
-    err = PyRun_SimpleString(buf);
-    assert(err == 0);
-    sprintf(buf, "sys.path.prepend('')");
+#if defined(__linux__) || defined(DARWIN)
+    // If /where/installed/lib/python/neuron exists, then append to sys.path
+    std::string lib = std::string(path_prefix_to_libnrniv());
+#else
+    std::string lib = std::string(neuronhome_forward()) + std::string("/lib");
+#endif
+    if (isDirExist(lib + std::string("python/neuron"))) {
+      std::string cmd = std::string("sys.path.append('") + lib + std::string("python')");
```

Comment:
is this the best way to concatenate strings?

Reply:
looks fine to me!
> (it's also sufficient to have std::string("a") + "b")

Reply:
Using append is more efficient (if not the most since it avoids some temporaries): 
 
```
std::string cmd = std::string("sys.path.append('").append(lib).append("python')");
```

Reply:
Thanks. As efficiency is not an issue here, I've simplified the concatentation according to Pramod's comment. I typically use sprintf for this kind of concatenation but thought I'd come into the 21st century for this and avoid figuring out how much char* to allocate and then delete.